### ISSUE TITLE
Add ability to Fail the pipeline if the code coverage was decreased

### ISSUE THREAD

**Issue by nachg** [cicirello/jacoco-badge-generator#21]
I want to implement the code coverage check as a required verification step for the PR's to the master branch. 
I think that it is possible to implement it as an additional feature for the **jacoco-badge-generator**.
example:

```
      - name: Generate JaCoCo Badge
        id: jacoco
        uses: cicirello/jacoco-badge-generator@v2.1.0
        with:
          fail-on-decrese: true
```

Also, it will be great to have such option as:
```
    fail-when-coverage-lessThan: 60
```

Which will fail the pipeline when coverage is less than provided argument.

**Comment by nachg** [comment]
I believe that it can be done by the **git diff** command. We need just to pars its output for the .csv file.

**Comment by cicirello** [comment]
Both of these are doable.  I have a question about the need for one of them though, specifically fail-when-coverage-lessThan.  That can actually be done by jacoco itself.  See for example, the documentation of this functionality for [maven](https://www.jacoco.org/jacoco/trunk/doc/check-mojo.html).  I also tracked down the equivalent for [gradle](https://docs.gradle.org/4.0/userguide/jacoco_plugin.html#sec:jacoco_report_violation_rules).

Is there a benefit to handling that separate from jacoco?  The only benefit I can see (possibly), unless I'm missing something, is to get to the badge step. But if the badge step fails the run (instead of jacoco), you still won't get to the workflow step that commits it.

Anyway, this is easily added anyway.  

The fail on decrease should be easy enough as well. And as far as I'm aware, that is not something already supported by jacoco.

**Comment by cicirello** [comment]
@nachg I just released a new version with the suggested ways of failing workflows (coverage decreases as well as if coverage is below a chosen threshold). See the readme for the 4 new inputs (2 inputs for coverage, and 2 for branches coverage). The on decrease inputs default to false (so existing workflows don't surprise users with failures), and the fail if less than inputs default to 0 for the same reason. The version with the new inputs is v2.2.0, but I also recently added a v2 tag (with a workflow to automatically keep it up to date), so users don't need to edit workflows to get any bug fixes, or other non-breaking changes.

Thanks for the suggestion. I might actually start using the `fail-on-coverage-decrease` feature on pull-requests in my projects.
